### PR TITLE
Allow peer force removal

### LINKED ISSUE #1137: [replication] Allow to force delete peer
Force delete semantics - delete even if peer contains local shards.

### PR DESCRIPTION
**Pull request by e-ivkov**
Closes #1137

### CODE REVIEW

File: lib/storage/src/content_manager/mod.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ pub mod consensus_ops {
     #[derive(Debug, Deserialize, Serialize, PartialEq, Eq, Hash, Clone)]
     pub enum ConsensusOperations {
         CollectionMeta(Box<CollectionMetaOperations>),
-        AddPeer(PeerId, String),
```
Comment (generall) [inline lib/storage/src/content_manager/mod.rs:34]: It will break the serialization compatibility, won't it?

Comment (e-ivkov) [reply lib/storage/src/content_manager/mod.rs:34]: Yes, it probably will. I can remove this change if we want to maintain compatibility

**Review by generall — APPROVED** [review]

File: src/actix/api/cluster_api.rs [hunk 2]
```diff
@@ -14,19 +21,33 @@ async fn cluster_status(dispatcher: web::Data<Dispatcher>) -> impl Responder {
...
+    dispatcher: web::Data<Dispatcher>,
+    peer_id: web::Path<u64>,
+    web::Query(force): web::Query<Force>,
+) -> impl Responder {
     let timing = Instant::now();
     let dispatcher = dispatcher.into_inner();
     let peer_id = peer_id.into_inner();
 
-    if dispatcher.peer_has_shards(peer_id).await {
+    let (has_shards, has_the_only_shard) = dispatcher.peer_has_shards(peer_id).await;
```
Comment (agourlay) [inline src/actix/api/cluster_api.rs:33]: Is this guarding validation protecting us from the case where two remove_peer calls are made in parallel?

Comment (e-ivkov) [reply src/actix/api/cluster_api.rs:33]: This is for handling the case where there is only 1 shard in collection - and therefore we cannot delete this shard. I added a comment below

**Pull request merged**